### PR TITLE
Solved 5 issues/feature-requests

### PR DESCRIPTION
**Pull request by melwyncarlo**
**My contributions:**

1. Solved issue #1400 
2. Solved issue #1395 
3. Added the ability to safely discard a block upon deletion
4. Solved issue #1396
5. Solved issue #1368

Regarding point number 3, earlier upon deleting a given block, you could not create a new block bearing the same name; well, now you can.

Regarding point number 2, the code talks about _splines_, but bears the equation of a _quadratic Bézier curve_.
Also, there is an analytic equation (which is generic) to compute the n-th degree Bézier curve, but the author of the code uses a very different approach (which is not compatible with the aforementioned analytic equation). 

Regarding point number 4, BOTH the bugs mentioned in the issue have been solved.

Regarding point number 5, the icon's size is implemented as is. The big mouse icon (located at `librecad/res/icons`) size was 80x80 pixels. The small mouse icon (located at `librecad/res/ui`) size is 24x24 pixels. I think _this_ icon was used before. To implement the right icon, just ensure that the icon size is about 25x25 pixels.

### CODE REVIEW

**Comment by melwyncarlo** [comment]
Argh! **Solved issue LibreCAD#1396** must be a separate pull request. Don't know how this happened.

File: librecad/src/actions/rs_actionblocksremove.cpp [hunk 1]
```diff
@@ -78,8 +78,9 @@ void RS_ActionBlocksRemove::trigger() {
 					if (e->rtti()==RS2::EntityInsert) {
 						RS_Insert* ins = (RS_Insert*)e;
 						if (ins->getName()==block->getName() && !ins->isUndone()) {
-                            document->addUndoable(ins);
+                            document->removeUndoable(ins);
```
Comment (melwyncarlo) [inline librecad/src/actions/rs_actionblocksremove.cpp:81]: These changes are w.r.t. point number 1.

File: librecad/src/lib/engine/lc_splinepoints.cpp [hunk 2]
```diff
@@ -999,7 +999,9 @@ RS_Vector LC_SplinePoints::GetSplinePointAtDist(double dDist, int iStartSeg,
 
 	if(dDist <= dQuadDist)
 	{
-		double dt = GetQuadPointAtDist(vStart, vControl, vEnd, 0.0, dDist);
+        double t0 = 0.0;
```
Comment (melwyncarlo) [inline librecad/src/lib/engine/lc_splinepoints.cpp:1002]: These changes are w.r.t. point number 2.

File: librecad/src/lib/engine/rs_dimangular.cpp [hunk 3]
```diff
@@ -131,6 +131,7 @@ RS_Entity* RS_DimAngular::clone() const
     d->setOwner( isOwner());
     d->initId();
     d->detach();
+    d->updateDim();
```
Comment (melwyncarlo) [inline librecad/src/lib/engine/rs_dimangular.cpp:134]: These changes are w.r.t. point number 4.

File: librecad/src/lib/engine/rs_dimdiametric.cpp [hunk 4]
```diff
@@ -76,6 +76,7 @@ RS_Entity* RS_DimDiametric::clone() const {
 	d->setOwner(isOwner());
 	d->initId();
 	d->detach();
+   d->updateDim();
```
Comment (melwyncarlo) [inline librecad/src/lib/engine/rs_dimdiametric.cpp:79]: These changes are w.r.t. point number 4.

File: librecad/src/lib/engine/rs_dimlinear.cpp [hunk 5]
```diff
@@ -83,6 +83,7 @@ RS_Entity* RS_DimLinear::clone() const {
 	d->setOwner(isOwner());
 	d->initId();
 	d->detach();
+   d->updateDim();
```
Comment (melwyncarlo) [inline librecad/src/lib/engine/rs_dimlinear.cpp:86]: These changes are w.r.t. point number 4.

File: librecad/src/lib/engine/rs_dimradial.cpp [hunk 6]
```diff
@@ -74,6 +74,7 @@ RS_Entity* RS_DimRadial::clone() const {
 	d->setOwner(isOwner());
 	d->initId();
 	d->detach();
+    d->updateDim();
```
Comment (melwyncarlo) [inline librecad/src/lib/engine/rs_dimradial.cpp:77]: These changes are w.r.t. point number 4.

File: librecad/src/main/qc_applicationwindow.cpp [hunk 7]
```diff
@@ -28,7 +28,7 @@
 **********************************************************************/
 
 // Changes: https://github.com/LibreCAD/LibreCAD/commits/master/librecad/src/main/qc_applicationwindow.cpp
-
+#include <iostream>
```
Comment (melwyncarlo) [inline librecad/src/main/qc_applicationwindow.cpp:31]: The _iostream_ header may be removed. I was using this header while performing tests.

File: librecad/src/main/qc_applicationwindow.cpp [hunk 8]
```diff
@@ -606,17 +606,24 @@ int QC_ApplicationWindow::showCloseDialog(QC_MDIWindow * w, bool showSaveAll)
...
-		a_map["FileSaveAs"]->setText(tr("Save &as..."));
-	}
-	else {
-		QString name = format_filename_caption(w->getDocument()->getFilename());
-		a_map["FileSave"]->setText(tr("&Save %1").arg(name));
-		a_map["FileSaveAs"]->setText(tr("Save %1 &as...").arg(name));
-	}
-	a_map["FileSave"]->setEnabled(w);
-	a_map["FileSaveAs"]->setEnabled(w);
+   if (windowIsInBlockMode) {
```
Comment (melwyncarlo) [inline librecad/src/main/qc_applicationwindow.cpp:609]: These changes are w.r.t. point number 1.

File: librecad/src/main/qc_applicationwindow.h [hunk 9]
```diff
@@ -311,6 +311,8 @@ public slots:
 
 private:
 
+    bool windowIsInBlockMode = false;
```
Comment (melwyncarlo) [inline librecad/src/main/qc_applicationwindow.h:314]: These changes are w.r.t. point number 1.

**Review by melwyncarlo — COMMENTED** [review]
4 issues/feature-requests solved and tested.

File: librecad/src/ui/forms/qg_mousewidget.cpp [hunk 10]
```diff
@@ -40,7 +40,7 @@ QG_MouseWidget::QG_MouseWidget(QWidget* parent, const char* name, Qt::WindowFlag
 
     lLeftButton->setText("");
     lRightButton->setText("");
-    lMousePixmap->setPixmap( QPixmap(":/icons/mouse.svg"));
+    lMousePixmap->setPixmap( QPixmap(":/ui/mousebuttons24.png"));
```
Comment (melwyncarlo) [inline librecad/src/ui/forms/qg_mousewidget.cpp:43]: These changes are w.r.t. point number 5.

**Comment by lordofbikes** [comment]
This has become a bit messy now.
I can't merge this easy, because there are some issues I may address later.
Of course, I can cherry-pick what's usable yet, but this cheats your credits, because you will not have the merged badge on this pull request. 

> Argh! Solved issue #1396 must be a separate pull request. Don't know how this happened.

If you don't know, you should read [Propose changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests) in GitHub Docs for best practices.
And probably git docs about [branching](https://git-scm.com/docs/git-branch).

Basically, you should avoid to push to your forks master branch. Create a branch for each issue, then you can create separate pull requests, one for each feature/issue. When a branch is merged, you can delete it in your fork and sync with upstream/master to pull the changes.
I probably should have write access to your fork, so I may split these changes for you, if you want me to do that.
But you should not have any local changes then, because you have to force update your local repo later, which will discard local changes.

I'd like to address my issues when the changes are split, because I had no detailed look at them now.
Just for #1400, I'm not sure if this will be ever merged. This feature request has too much impact on long-time users and neither is there a decision made nor is the feature discussed adequate for a smooth implementation.

You're welcome to join our Zulip or IRC chat, both linked on https://librecad.org/, for further questions or to discuss contributions in advance.

**Comment by melwyncarlo** [comment]
I have rearranged all my pull-requests separately.

I agree that it's messy. I have never submitted multiple pull-requests one after the other before (without the previous ones being merged first). That is why I had encountered the above problem; it's my first multiple pull-request.

Thank you very much for your help and guidance on this matter. 

Regarding #1400, I only worked on that because of the `feature request` label appended to it. Anyway, it's fine.

Thank you very much for the invite. I have just joined the LibreCAD group on Zulip.

As this pull-request is now redundant, I will close it.

**Pull request closed without merging**